### PR TITLE
Implement message scroll when height changes

### PR DESCRIPTION
**Pull request by sravan-s**

### CODE REVIEW

File: src/components/CustomMessage.tsx [hunk 1]
```diff
@@ -46,6 +46,31 @@ export default function CustomMessage(props: Props) {
...
+          lastMessageRef?.current?.scrollIntoView({behavior: 'smooth'});
+        } catch (e) {
+          console.error(e);
+        }
+      }
+    });
+    if (isLastBotMessage && lastMessageRef?.current) {
+      const targetNode = lastMessageRef?.current;
+      // create mutation observer
+      observer.observe(targetNode);
```
Comment (sravan-s) [inline src/components/CustomMessage.tsx:68]: attach conditionally maybe? ie -> when the last message is visible on screen (?)

File: src/css/index.css [hunk 2]
```diff
@@ -1,3 +1,6 @@
+.sb-msg--bot {
```
Comment (sravan-s) [inline src/css/index.css:1]: useless

File: src/components/CustomMessage.tsx [hunk 3]
```diff
@@ -98,7 +123,12 @@ export default function CustomMessage(props: Props) {
     }
   });
 
-  return <div>
+  return <div
+    ref={lastMessageRef}
+    className={
+      isBotMessage ? 'sb-msg--bot' : 'sb-msg--user'
+    }
```
Comment (sravan-s) [inline src/components/CustomMessage.tsx:130]: useless

File: src/components/CustomMessage.tsx [hunk 4]
```diff
@@ -46,6 +46,31 @@ export default function CustomMessage(props: Props) {
...
+  const isLastBotMessage: boolean = isBotMessage && allMessages[allMessages.length - 1].messageId === message.messageId;
+  const lastMessageRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const observer = new ResizeObserver((entries) => {
+      for (const entry of entries) {
+        const newHeight = entry.contentRect.height;
+        try {
+          console.log(`New height: ${newHeight}px`);
+          lastMessageRef?.current?.scrollIntoView({behavior: 'smooth'});
```
Comment (sravan-s) [inline src/components/CustomMessage.tsx:59]: throttle this maybe?

**Review by AhyoungRyu — APPROVED** [review]
Shall we go? 🚀 
record

File: src/components/CustomMessage.tsx (hunk 4 above)
Comment (AhyoungRyu) [reply src/components/CustomMessage.tsx]: Added https://github.com/sendbird/ai-bot-url-webdemo/pull/27/commits/9c02e4b155438acf798c03796ef4ec89693e629e but couldn't set a larger number of delay cause it made the scroll interaction a bit slow.

File: src/components/CustomMessage.tsx [hunk 5]
```diff
@@ -46,6 +47,40 @@ export default function CustomMessage(props: Props) {
   const firstMessageId = firstMessage?.messageId ?? -1;
   const demoStates = useContext<DemoConstant>(DemoStatesContext);
   const isWebDemo: boolean = demoStates.name === 'webDemo';
+  const isBotMessage: boolean = (message as UserMessage).sender.userId === botUser.userId;
+  const isLastBotMessage: boolean = isBotMessage && allMessages[allMessages.length - 1].messageId === message.messageId;
+  const lastMessageRef = useRef<HTMLDivElement>(null);
+
+  const throttledScrollIntoView = useThrottle(
+    (element: HTMLDivElement) => {
+      element.scrollIntoView({ behavior: "smooth", block: "end" });
```
Comment (AhyoungRyu) [inline src/components/CustomMessage.tsx:56]: Adding `block: 'end'` solved the unexpectedly stopping scroll in the middle.

**Pull request merged**